cedar-policy-validator/src/json_schema.rs

```diff
@@ -1314,6 +1326,163 @@ impl<N> From<TypeVariant<N>> for Type<N> {
     }
 }
 
+/// The fields for a `Type`. Used for implementing deserialization.
```

Comment:
Why duplicate this for annotated types? Do we still need to be able to parse types without annotations?

Reply:
I believe so. `Type<N>` is used as a value of `shape`. We wouldn't want something like,
```JSON
"shape": {
  "annotations": {"a": ""},
  "type": "Foo"
}
```

Reply:
Agreed. I tried to eliminate the duplication like the following,

```rust
    fn visit_map<A>(self, mut map: A) -> std::result::Result<Self::Value, A::Error>
    where
        A: MapAccess<'de>
    {
        let mut pairs: Vec<(String, serde_json::Value)> = Vec::new();
        let mut annotations: Option<Annotations> = None;

        while let Some(key) = map.next_key::<String>()? {
            if key == "annotations" {
                annotations = Some(map.next_value()?);
            } else {
                pairs.push((key, map.next_value()?));
            }
        }
        let rest: MapDeserializer<'_, std::vec::IntoIter<(String, serde_json::Value)>, A::Error> = MapDeserializer::new(pairs.into_iter());
        let ty_visitor = TypeVisitor {
            _phantom: PhantomData,
        };
        Ok(Self::Value {
            data: rest.deserialize_any(ty_visitor)?,
            annotations: annotations.unwrap_or_default(),
        })
    }
```

But I couldn't get type checker to pass.

Reply:
If it means we can simplify the code could we consider revising the RFC allowing? I doesn't seem too bad to me

```json
"shape": {
  "annotations": {"a": ""},
  "type": "Foo"
}
```

i guess in the Cedar format it would be

```
entity foo 
@a("")
{
...
}
```

Reply:
let's be faithful to the RFC.